openmc/universe.py

```diff
@@ -396,6 +400,36 @@ def plot(self, origin=(0., 0., 0.), width=(1., 1.), pixels=(200, 200),
                 height = pixels[0]*px/(params.top - params.bottom)
                 fig.set_size_inches(width, height)
 
+            # add legend showing which colors represent which material
+            # or cell if that was requested
+            if legend:
+                if plot.colors is None:
+                    raise Exception("Must set plot color dictionary if you would"
+                                    " like to have an automatic legend.")
+
+                if color_by == "cell":
+                    expected_key_type = openmc.Cell
+                elif color_by == "material":
+                    expected_key_type = openmc.Material
+                else:
+                    raise Exception("wtf?")
```

Comment:
```suggestion
                else:
                    expected_key_type = openmc.Material
```

The color_by setter in plot.py has a cv.check_value that guards against incorrect color_by values so perhaps we can consider removing some lines here

https://github.com/openmc-dev/openmc/blob/ae615c02d9bb97f05f52a6b74494f1dc37514186/openmc/plots.py#L301-L304